RF: Lib introduced to consolidate the pygfx imports

maharshi-gor opened the pull request:
### RF: Lib introduced to consolidate the pygfx imports

**PR Contents**

- Removes the `io` folder created for different io and flatten it to `io.py`.
- Update imports and introduce the `lib.py` file to consolidate the pygfx imports throughout the project.

pep8speaks commented:
Hello @maharshi-gor! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`fury/io.py`](https://github.com/fury-gl/fury/blob/c82c2b7be47de2fcff9906c07c76e6936cbda1bc/fury/io.py):

> [Line 51:80](https://github.com/fury-gl/fury/blob/c82c2b7be47de2fcff9906c07c76e6936cbda1bc/fury/io.py#L51): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)

skoudoro commented on fury/animation/animation.py:
```diff
@@ -1,1446 +1,1454 @@
-from collections import defaultdict
```
Why do you comment all this file ? it just needs few update /few comments.... 

I do not understand. most of it do not use vtk...

skoudoro commented on fury/io.py:
```diff
@@ -1,40 +1,77 @@
+        The display extent (width, height, depth), by default None
+    generate_mipmaps : bool, optional
+        automatically generates mipmaps when transferring data to the GPU,
+        by default True
```
default to remove

skoudoro commented on fury/io.py:
```diff
@@ -1,40 +1,77 @@
+    fnames : list
+        filenames to generate cube map.
+    size : tuple, optional
+        The display extent (width, height, depth), by default None
```
default to remove

skoudoro reviewed (commented):
Hi @maharshi-gor,

See below an initial review. Do not forget to add test for any new functions. Let's not forget this habit.

maharshi-gor commented on fury/__init__.pyi:
```diff
@@ -4,58 +4,59 @@
-#     "utils",
-#     "window",
-# ]
+__all__ = [
+    #     "actor",
+    #     "actors",
+    #     "animation",
+    #     "colormap",
+    #     "convert",
+    #     "data",
+    #     "deprecator",
+    #     "decorators",
+    #     "gltf",
+    #     "interactor",
+    #     "io",
+    #     "layout",
+    "lib",
+    #     "material",
+    #     "molecular",
+    #     "optpkg",
+    #     "pick",
+    #     "pkg_info",
+    #     "primitive",
+    #     "shaders",
+    #     "stream",
+    #     "testing",
+    #     "transform",
+    #     "ui",
+    #     "utils",
+    #     "window",
```
I will update for `io` but `actor` is yet to be initiated

skoudoro replied:
check all of them please, not only `io`. It is ok for many of them

maharshi-gor commented on fury/__init__.pyi:
```diff
@@ -4,58 +4,59 @@
-#     window,
-# )
+from . import (
+    #     actors,
+    #     animation,
+    #     colormap,
+    #     convert,
+    #     data,
+    #     decorators,
+    #     deprecator,
+    #     gltf,
+    #     interactor,
+    #     io,
+    #     layout,
+    #     optpkg,
+    #     pkg_info,
+    #     primitive,
+    #     shaders,
+    #     stream,
+    #     testing,
+    #     ui,
+    #     window,
```
I will update for `io` but `actor` is yet to be initiated

skoudoro replied:
check all of them please, not only `io`. It is ok for many of them

maharshi-gor commented on fury/animation/__init__.py:
```diff
@@ -1,3 +1,3 @@
-import lazy_loader as lazy
+# import lazy_loader as lazy
```
This started because the code format was not working as it was part of some chain errors once it has been tested and working.

maharshi-gor replied:
`from fury.lib import Actor, Camera, Transform`

This import is not valid as we removed `vtk` from the animation file. Due to this all the files using it needed to be commented out.

skoudoro replied:
Please, just comment this specific line and where it is use. 

another solution could be to define them in `fury.lib`,  you could do:
```
Actor = object
Camera = object  # check the camera in pygfx
Transform = object # check the transform in pygfx

maharshi-gor commented on fury/animation/__init__.pyi:
```diff
@@ -1,51 +1,51 @@
-    "tan_cubic_spline_interpolator",
-    "xyz_color_interpolator",
-]
+# __all__ = [
+#     "Animation",
+#     "CameraAnimation",
+#     # "Timeline",
+#     "euclidean_distances",
+#     "get_next_timestamp",
+#     "get_previous_timestamp",
+#     "get_time_tau",
+#     "get_timestamps_from_keyframes",
+#     "get_values_from_keyframes",
+#     "lerp",
+#     "color_interpolator",
+#     "cubic_bezier_interpolator",
+#     "cubic_spline_interpolator",
+#     "hsv_color_interpolator",
+#     "lab_color_interpolator",
+#     "linear_interpolator",
+#     "slerp",
+#     "spline_interpolator",
+#     "step_interpolator",
+#     "tan_cubic_spline_interpolator",
+#     "xyz_color_interpolator",
+# ]
```
# from fury.lib import Actor, Camera, Transform

This import is not valid as we removed `vtk` from the animation file. Due to this all the files using it needed to be commented out.

maharshi-gor replied:
I will tackle the specific pygfx class by incorporating animation pr in the actor PR.

I have commented on erroneous parts.

maharshi-gor commented:
I have added some tests for the `io.py` file, and `test_io.py` is running successfully. you can test with `pytest fury/tests/test_io.py` to run the tests.

I have also added testing file back to enable the testing of some functionality which might be required.

skoudoro commented:
Thank you for the update merging
